### DISCUSSION TITLE
Help me figure out architeture

### DISCUSSION THREAD

**Discussion by alpharde** [seaweedfs/seaweedfs#3137] [Q&A]
Ok, so I've tried to create a decent test environment with seaweedfs only to fail miserably a couple of times. I tried to simulate outages to test resiliency and sometimes ended up with broken data.

Here's what I have:
2 homelab servers with 3x 500GB each
1 server in the cloud, also with 3x 500GB HDDs
couple of cloud VMs with large storage

What I want:
No SPOF
Using LXD over posix mountpoint
Good upgrade advice for adding more storage later and keeping it no spof

I need some help figuring out a good no spof strategy, namely one that resists one home server being down (for maintenance) or the two being shot at the same time (internet connection off, power outage). Do I need something else? I could add more machines locally, also if having cloud servers mixed up somehow feel stupid I can take that out too.
And about LXD, it basically stores plain files on a directory and binds the directory somewhere else (/var/lib/lxd/containers ...), basic stuff I believe.

**Comment by chrislusf** [comment]
you can setup one cloud cluster and one home cluster, and sync them.

**Comment by alpharde** [comment]
How I should sync then? Does this means that the daemons (master/volume/etc) wouldn't be connected?

**Reply by chrislusf** [reply]
You can use `weed filer.sync` to sync the clusters. https://github.com/chrislusf/seaweedfs/wiki/Filer-Active-Active-cross-cluster-continuous-synchronization

You can use `weed filer.backup` to backup. https://github.com/chrislusf/seaweedfs/wiki/Async-Backup

You can also use https://github.com/chrislusf/seaweedfs/wiki/Cloud-Drive-Architecture to just backup to a cloud object storage.
